matching/src/main/scala/org/kframework/backend/llvm/matching/Matrix.scala

```diff
@@ -549,6 +549,42 @@ class Matrix private(val symlib: Parser.SymLib, private val rawColumns: IndexedS
     }
   }
 
+  private def useless(r: Row): Boolean = {
+    if (columns.size == 0) {
+      if (rows.size > 0) {
+        false
+      } else {
+        true
+      }
+    }
+
+    if (!r.patterns(0).isWildcard) {
+      val con = r.patterns(0).signature(r.clause).head
+      val rowSpec = Matrix.fromRows(symlib, IndexedSeq(r), fringe).specialize(con, 0)._2.rows.head
```

Comment:
let's add a specialize function to Row.

Reply:
It should probably take a Constructor and a column index and return an Option[Row] which is None if the row was deleted during the specialization.